packages/nodejs/src/__tests__/transmitter.test.ts

```diff
@@ -0,0 +1,168 @@
+          on: () => {},
+          write: () => {},
+          end: () => {}
+        }
+
+        return (clientRequestMock as unknown) as ClientRequest
+      })
+    }
+
+    it("resolves to a response on success", async () => {
+      const transmitter = sampleTransmitter()
+
+      mockSampleRequest()
+
+      await expectSampleResponse(transmitter.transmit())
+    })
+
+    it("resolves to an empty object if the response is not JSON", async () => {
+      const transmitter = sampleTransmitter()
+
+      mockSampleRequest({ responseBody: "not JSON" })
+
+      await expectSampleResponse(transmitter.transmit(), {})
+    })
+
+    it("rejects on error", async () => {
+      const transmitter = sampleTransmitter()
+
+      // do not mock the request -- beforeAll uses nock to disable network
+      // requests, so any request to a non-mocked host will throw an error
+
+      await expect(transmitter.transmit()).rejects.toMatchObject({
+        error: expect.any(Error)
+      })
+    })
+
+    it("uses config values in the query string", async () => {
+      process.env.APPSIGNAL_PUSH_API_KEY = "some_api_key"
+      process.env.APPSIGNAL_APP_NAME = "myApp"
+      ;(process.env as any).NODE_ENV = "development"
+      process.env.APPSIGNAL_HOSTNAME = "myHostname"
+
+      const transmitter = sampleTransmitter()
+
+      mockSampleRequest({
+        query: {
+          api_key: "some_api_key",
+          name: "myApp",
+          environment: "development",
+          hostname: "myHostname"
+        }
+      })
+
+      await expectSampleResponse(transmitter.transmit())
+    })
+
+    it("uses the CA file from the config", async () => {
+      // disable nock, so we can mock the https library ourselves
+      nock.restore()
+
+      process.env.APPSIGNAL_CA_FILE_PATH = "/foo/bar"
+
+      jest.spyOn(fs, "accessSync").mockImplementation(() => {})
+
+      const fsReadFileSyncMock = jest
+        .spyOn(fs, "readFileSync")
+        .mockImplementation(() => "ca file contents")
+
+      const httpsRequestMock = mockHTTPSRequest()
+
+      sampleTransmitter().transmit()
+
+      expect(fsReadFileSyncMock).toHaveBeenCalledWith("/foo/bar", "utf-8")
+
+      expect(httpsRequestMock).toHaveBeenCalledWith(
+        expect.objectContaining({ ca: "ca file contents" }),
+        expect.anything()
+      )
+    })
+
+    it("uses the system CAs if the CA file is not readable", () => {
+      // disable nock, so we can mock the https library ourselves
+      nock.restore()
+
+      process.env.APPSIGNAL_CA_FILE_PATH = "/foo/bar"
+
+      jest.spyOn(fs, "accessSync").mockImplementation(path => {
+        if (path == "/foo/bar") {
+          throw new Error("No CA for you!")
+        }
+      })
+
+      const consoleWarnSpy = jest.spyOn(console, "warn")
+
+      const httpsRequestMock = mockHTTPSRequest()
+
+      sampleTransmitter().transmit()
+
+      expect(httpsRequestMock).toHaveBeenCalledWith(
+        expect.objectContaining({ ca: "" }),
```

Comment:
Have we actually tested it works with an empty string? It maybe better if we not set the `ca` property when no file could be read?

Reply:
Last time @luismiramirez and I were looking into the transmitter, we did check that an empty string value was interpreted the same way as if the property was not set.

I do agree not setting it at all would be preferable, and clearer -- I'll see if the implementation can be easily changed to do that.

Reply:
Done! Now the property is not set.

Reply:
Awesome! 🚀